lib/lintPostcssResult.mjs

```diff
@@ -107,13 +108,16 @@ export default async function lintPostcssResult(stylelintOptions, postcssResult,
 			},
 		};
 
-		performRules.push(
-			Promise.all(
-				postcssRoots.map((postcssRoot) =>
-					ruleFunction(primaryOption, secondaryOptions, context)(postcssRoot, postcssResult),
-				),
-			),
-		);
+		const ruleFn = ruleFunction(primaryOption, secondaryOptions, context);
+
+		/**
+		 * @param {import('postcss').Root} postcssRoot
+		 */
+		async function timedRuleFn(postcssRoot) {
+			return await timing.time(ruleName, async () => ruleFn(postcssRoot, postcssResult))();
+		}
+
+		performRules.push(Promise.all(postcssRoots.map(timedRuleFn)));
```

Comment:
question

Wouldn't it be better to use `timing.time` only if necessary?
e.g.

```js
const foo = async (postcssRoot) => ruleFn(postcssRoot, postcssResult);

/* … */

const fn = process.env.TIMING ? timedRuleFn : foo;
performRules.push(Promise.all(postcssRoots.map(fn)));
```

Reply:
Thanks for catching it!
I wanted to encapsulate the env access related to timing within `timing.js`, so I tried it like this.
[Changed to run only when timing is enabled](https://github.com/stylelint/stylelint/pull/8108/commits/ea546fb59c699ae9349031003c986735afe0b83d)